ParProcCo/msmapper_aggregator.py

```diff
@@ -201,25 +205,35 @@ def _get_all_axes(self) -> None:
     def _accumulate_volumes(self) -> None:
         for data_file, slices in zip(self.output_data_files, self.all_slices):
             with h5py.File(data_file, "r") as f:
-                volumes = []
-                volumes = [np.array(f[self.nxdata_path_name][self.signal_name])]
+                aux_signals = []
+                volume = np.array(f[self.nxdata_path_name][self.signal_name])
                 if self.renormalisation:
                     weights = np.array(f[self.nxdata_path_name]["weight"])
-                if self.accumulate_aux_signals:
-                    extra_signal_names = [name for name in self.aux_signal_names if name != "weight"]
-                    volumes += [np.array(f[self.nxdata_path_name][name]) for name in extra_signal_names]
+                if self.non_weight_aux_signal_names:
+                    for name in self.non_weight_aux_signal_names:
+                        aux_signals.append(np.array(f[self.nxdata_path_name][name]))
 
             if self.renormalisation:
-                volumes = [np.multiply(volume, weights) for volume in volumes]
+                volume = np.multiply(volume, weights)
                 self.accumulator_weights[slices] += weights
-            for volume in volumes:
-                self.accumulator_volume[slices] += volume
+                aux_signals = [np.multiply(aux_signal, weights) for aux_signal in aux_signals]
+
+            self.accumulator_volume[slices] += volume
+
+            for count, aux_signal in enumerate(aux_signals):
+                self.accumulator_aux_signals[count][slices] += aux_signal
 
         if self.renormalisation:
             self.accumulator_volume = self.accumulator_volume / self.accumulator_weights
+            if self.non_weight_aux_signal_names:
+                self.accumulator_aux_signals = self.accumulator_aux_signals / self.accumulator_weights
```

Comment:
Don't think this works - needs to be an explicit loop. Add unit test with more than one non-weight aux signal